hid_enabled.go

```diff
@@ -226,3 +359,39 @@ func (dev *Device) Read(b []byte) (int, error) {
 	}
 	return read, nil
 }
+
+// Type identify the device as a HID device
+func (dev *HidDevice) Type() DeviceType {
+	return dev.DeviceInfo.Type()
+}
+
+// GenericDeviceHandle represents a libusb device_handle struct
+type GenericDeviceHandle *C.struct_libusb_device_handle
+
+// GenericLibUsbDevice represents a libusb device struct
+type GenericLibUsbDevice *C.struct_libusb_device
+
+// GenericDeviceOpen is a helper function to call the C version of open.
+func GenericDeviceOpen(dev GenericLibUsbDevice) (GenericDeviceHandle, error) {
+	var handle GenericDeviceHandle
+	errCode := int(C.libusb_open(dev, (**C.struct_libusb_device_handle)(&handle)))
+	if errCode < 0 {
+		return nil, fmt.Errorf("Error opening generic USB device %v, code %d", dev, errCode)
+	}
+	return handle, nil
+}
+
+// GenericDeviceClose is a helper function to close a libusb device
+func GenericDeviceClose(handle GenericDeviceHandle) {
+	C.libusb_close(handle)
+}
+
+// InterruptTransfer is a helpler function for libusb's interrupt transfer function
+func InterruptTransfer(handle GenericDeviceHandle, endpoint uint8, data []byte, timeout uint) ([]byte, error) {
```

Comment:
Please document the unit of `timeout` or make it `time.Duration` and convert to libusb units.

Reply:
I have unexported this function and as a result removed the parameter altogether, it's not used by this library anyway.